modules/document_repository/jsx/editForm.js

```diff
@@ -0,0 +1,251 @@
+/* exported DocEditForm */
+import PropTypes from 'prop-types';
+/**
+ * Document Edit Form
+ *
+ * Fetches data corresponding to a given file from Loris backend and
+ * displays a form allowing meta information of the media file
+ *
+ * @author Shen Wang
+ * @version 1.0.0
+ *
+ * */
+class DocEditForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      Data: {},
+      docData: {},
+      uploadResult: null,
+      isLoaded: false,
+      loadedData: 0,
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.showAlertMessage = this.showAlertMessage.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  fetchData() {
+    return fetch(this.props.DataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => {
+      let formData = {
+          idDocFile: data.docData.id,
+          pscid: data.docData.pscid,
+          category: data.docData.category,
+          visitLabel: data.docData.visitLabel,
+          comments: data.docData.comments,
+          version: data.docData.version,
+          instrument: data.docData.instrument,
+          category: data.docData.category,
+          forSite: data.docData.forSite,
```

Comment:
Since all these indexes match, you can just do `const formData = data.docData`. The only index that is different is `idDocFile`, so you'll just have to adjust for that through the code.